third_party/2and3/google/protobuf/internal/containers.pyi

```diff
@@ -1,9 +1,27 @@
 from google.protobuf.descriptor import Descriptor
 from google.protobuf.internal.message_listener import MessageListener
+from google.protobuf.internal.python_message import GeneratedProtocolMessageType
 from google.protobuf.message import Message
-from typing import Sequence, TypeVar, Generic, Any, Iterator, Iterable, Union, Optional, Callable, overload, List
+from typing import (
+    Any,
+    Callable,
+    Generic,
+    Iterable,
+    Iterator,
+    List,
+    Mapping as tMapping,
+    MutableMapping as tMutableMapping,
```

Comment:
Since they are just aliased later, this should just be:

```suggestion
    Mapping as Mapping,
    MutableMapping as MutableMapping,
```

Reply:
This sounds reasonable to me as long as the types are re-exported.
`Mapping` and `MutableMapping` are types at global scope of the containers.py impl
https://github.com/protocolbuffers/protobuf/blob/master/python/google/protobuf/internal/containers.py

I'm haven't tested how mypy feels about re-exporting imports in that way.

Reply:
Using `from y import x as x` is mandated by PEP 484 to re-export items in stubs. This is supported by mypy.

Reply:
awesome - good to hear.
I've updated the diff with this technique. Much cleaner. Thanks for pointing it out.